### ISSUE TITLE
🐛  Incorrect line numbers in side by side view

### ISSUE THREAD

**Issue by jpalus** [dandavison/delta#818]
Unless I'm misinterpreting numbers on the left, it appears that line numbers in side by side view are incorrect:
![sbs-ln](https://user-images.githubusercontent.com/484155/144488322-57b94183-f4c4-476d-a5a5-8428a86bb99f.png)
Original diff:
```
commit df29ab3b7b6e7bd55c23460d76fa841010fd0084 (HEAD -> master, origin/master, origin/HEAD)
Author: Jan Palus <atler@pld-linux.org>
Date:   Thu Dec 2 19:54:09 2021

    up to 21.3.1

diff --git a/Mesa.spec b/Mesa.spec
index ac6f654..de5a919 100644
--- a/Mesa.spec
+++ b/Mesa.spec
@@ -71,14 +71,14 @@
 Summary:	Free OpenGL implementation
 Summary(pl.UTF-8):	Wolnodostępna implementacja standardu OpenGL
 Name:		Mesa
-Version:	21.3.0
+Version:	21.3.1
 Release:	1
 License:	MIT (core) and others - see license.html file
 Group:		X11/Libraries
 #Source0:	ftp://ftp.freedesktop.org/pub/mesa/mesa-%{version}.tar.xz
 ## Source0-md5:	7c61a801311fb8d2f7b3cceb7b5cf308
 Source0:	https://gitlab.freedesktop.org/mesa/mesa/-/archive/mesa-%{version}/mesa-mesa-%{version}.tar.bz2
-# Source0-md5:	3230a374056e951e73c3206de3453211
+# Source0-md5:	cc749bf46a05c88bdb7ba03dc53da6fd
 Patch0:		zink_x32.patch
 URL:		https://www.mesa3d.org/
 %{?with_opencl_spirv:BuildRequires:	SPIRV-LLVM-Translator-devel >= 8.0.1.3}
```
I would have expected that line numbers are exactly the same on both sides and continuous but on the left number 74 appears twice and from that place they don't match between sides.

delta 0.10.2
git 2.34.1

**Comment by dandavison** [comment]
Hi @jpalus, thanks very much for this. cc @th1000s `git bisect` indicates that the bug was introduced at fbb654d4370139b2d0fdef9c1b908fae9c54d797

**Comment by th1000s** [comment]
Nice find, shows that all the test diffs are too minimal. Just a single one with context would have caught this :)

**Comment by dandavison** [comment]
The fix for this is now released: `0.10.3`
